pwndbg/elf.py

```diff
@@ -68,6 +70,59 @@ def read(typ, address, blob=None):
     return obj
 
 
+@pwndbg.proc.OnlyWhenRunning
+def get_containing_segments(elf_filename, vaddr, loadaddr):
+    local_path = pwndbg.file.get_file(elf_filename)
+    segments = []
+    with open(local_path, 'rb') as f:
+        elffile = ELFFile(f)
+        is_pic = elffile.header['e_type'] == 'ET_DYN'
+        load = loadaddr if is_pic else 0
+        for seg in elffile.iter_segments():
+            if 'GNU_STACK' in seg['p_type']:
+                continue
+
+            if vaddr < load + seg['p_vaddr'] or vaddr >= load + seg['p_vaddr'] + seg['p_memsz']:
+                continue
+
+            x = dict(seg.header)
+            # add enriched attributes
+            flags = seg['p_flags']
+            x['x_perm_read'] = flags & PF_R != 0
+            x['x_perm_write'] = flags & PF_W != 0
+            x['x_perm_execute'] = flags & PF_X != 0
+            x['x_real_vaddr_start'] = load + x['p_vaddr']
+            x['x_real_vaddr_end'] = load + x['p_vaddr'] + x['p_memsz']
+            x['x_file_backing_end'] = x['x_real_vaddr_start'] + x['p_filesz']
+            x['x_is_file_backed'] = vaddr <= x['x_file_backing_end']
+            segments.append(x)
+    return segments
+
+
+@pwndbg.proc.OnlyWhenRunning
+def get_containing_sections(elf_filename, vaddr, loadaddr):
```

Comment:
We can probably make both functions to be `memoize.reset_on_objfile`

Reply:
Agreed. Can craft another commit for that